dojo_plugin/pages/workspace.py

```diff
@@ -20,88 +20,18 @@
-    message = "-".join(args).encode()
-    return hmac.HMAC(key, message, "sha256").hexdigest()
-
-def start_on_demand_service(user, service_name):
-    if service_name not in on_demand_services:
-        return
-    try:
-        exec_run(
-            f"/run/current-system/sw/bin/dojo-{service_name}",
-            workspace_user="hacker",
-            user_id=user.id,
-            assert_success=True,
-        )
-    except docker.errors.NotFound:
-        return False
-    return True
-
-@workspace.route("/workspace/desktop")
-@authed_only
-def view_desktop():
-    user_id = request.args.get("user")
-    password = request.args.get("password")
-    url_only = request.args.get("url_only")
-
-    if user_id and not password and not is_admin():
-        abort(403)
-
-    user = get_current_user() if not user_id else Users.query.filter_by(id=int(user_id)).first_or_404()
-    container = get_current_container(user)
-    if not container:
-        return "" if url_only else render_template("iframe.html", active=False)
-
-    interact_password = container_password(container, "desktop", "interact")
-    view_password = container_password(container, "desktop", "view")
-
-    if user_id and password:
-        if not hmac.compare_digest(password, interact_password) and not hmac.compare_digest(password, view_password):
-            abort(403)
-        password = password[:8]
-    else:
-        password = interact_password[:8]
-
-    view_only = user_id is not None
-    service = "~".join(("desktop", str(user.id), container_password(container, "desktop")))
-
-    vnc_params = {
-        "autoconnect": 1,
-        "reconnect": 1,
-        "reconnect_delay": 200,
-        "resize": "remote",
-        "path": url_for("pwncollege_workspace.forward_workspace", service=service, service_path="websockify"),
-        "view_only": int(view_only),
-        "password": password,
-    }
-    iframe_src = url_for("pwncollege_workspace.forward_workspace", service=service, service_path="vnc.html", **vnc_params)
-
-    share_urls = {
-        "Desktop (Interact)": url_for("pwncollege_workspace.view_desktop", user=user.id, password=interact_password, _external=True),
-        "Desktop (View)": url_for("pwncollege_workspace.view_desktop", user=user.id, password=view_password, _external=True),
-    }
-
-    if start_on_demand_service(user, "desktop") is False:
-        return render_template("iframe.html", active=False)
-
-    return iframe_src if url_only else render_template("iframe.html",
-                           iframe_name="workspace",
-                           iframe_src=iframe_src,
-                           share_urls=share_urls,
-                           active=True)
 
 @workspace.route("/workspace/<service>")
 @authed_only
 def view_workspace(service):
     user = get_current_user()
     active = bool(get_current_dojo_challenge())
+    iframe_src = "" if service=="desktop" else f"/workspace/{service}/"
+    loadUrlAsync = service == "desktop"
```

Comment:
Let's get rid of the decision to `loadUrlAsync`. All `/workspace/*` should be loaded async; the desktop is not special :)

Reply:
code just seams to have the iframe src /workspace/code/
what other workspaces are there ? :)